spec/ics-2-consensus-requirements/README.md

```diff
@@ -0,0 +1,76 @@
+---
+ics: 2
+title: Consensus Requirements
+stage: proposal
+category: ibc-core
+author: Juwoon Yun <joon@tendermint.com>, Christopher Goes <cwgoes@tendermint.com>
+created: 2019-02-25
+modified: 2019-03-05
+---
+
+The IBC protocol creates a mechanism by which two replicated fault-tolerant state machines may pass messages to each other. These messages provide a base layer for the creation of communicating blockchain architecture that overcomes challenges in the scalability and extensibility of computing blockchain environments.
+
+The IBC protocol assumes that multiple applications are running on their own blockchain with their own state and own logic. Communication is achieved over an ordered message queue primitive, allowing the creation of complex inter-chain processes without trusted third parties.
+
+The message packets are not signed by one psuedonymous account, or even multiple, as in multi-signature sidechain implementations. Rather, IBC assigns authorization of the packets to the source blockchain's consensus algorithm, performing light-client style verification on the destination chain. The Byzantine-fault-tolerant properties of the underlying blockchains are preserved: a user transferring assets between two chains using IBC must trust only the consensus algorithms of both chains.
+
+In order to achive this, a blockchain should be able to check 1. whether the block is valid and 2. whether the packet is included in the block or not. In this proposal, we introduce an abstract concept of blockchains, called `Header`. `Header` requires its consensus algorithm to satisfy some properties, including deterministic safety, lightclient compatibility and valid transition of the state machine.
+
+## Specification
+
+### Motivation
+
+`Header` defines required properties of the blockchain on the network. The implementors can check whether the consensus that they are using is qualified to be connected to the network or not. If not, they can modify the algorithm or wrap it with additional logic to make it compatible with the specification.
+
+### Desired Properties
+
+`Header` is `(p : Maybe<Header>, v : Header -> bool, s : []byte -> []byte)` where `p` is parent block, `v` is lightclient verifier and `s` is state. The other parts of the protocol, such as connections and channels, are defined over `Header`s, so if a blockchain wants to establish an IBC communication with another, it is required to satisfy the properties of `Header`. 
+
+Definitions:
+
+1. Direct children have their parent as `p` and are verifiable with their parent.
+2. The height of a block is the step of referring `p` required to get to `nil`
+
+Requirements:
+
+1. Headers have only one direct child
```

Comment:
If there is a chain with probabalistic finality using some BFT overlay to produce "checkpoints" at "epochs", then could we consider these "checkpoints" as Headers for the purposes of IBC, while the rest of the block headers do not satisfy the ibc `Header` property?

Reply:
Definitely. These definitions will cover the finality overlays on nondeterministic finality chains, such as pegzones, Casper FFG, etc.